packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts

```diff
@@ -431,50 +456,68 @@ export class AzureCommunicationCallWithChatAdapter implements CallWithChatAdapte
       attachmentMetadata?: AttachmentMetadata[];
     }
   ): Promise<void> {
-    return await this.chatAdapter.updateMessage(
-      messageId,
-      content,
-      metadata,
-      /* @conditional-compile-remove(attachment-download) @conditional-compile-remove(attachment-upload) */ options
-    );
+    return this.chatAdapterPromise.then((adapter) => {
+      return adapter.updateMessage(
+        messageId,
+        content,
+        metadata,
+        /* @conditional-compile-remove(attachment-download) @conditional-compile-remove(attachment-upload) */ options
+      );
+    });
   }
   /** Delete an existing message. */
   public async deleteMessage(messageId: string): Promise<void> {
-    return await this.chatAdapter.deleteMessage(messageId);
+    return await this.chatAdapterPromise.then((adapter) => {
+      return adapter.deleteMessage(messageId);
+    });
   }
   /* @conditional-compile-remove(attachment-download) @conditional-compile-remove(attachment-upload) */
   public registerActiveFileUploads = (files: File[]): FileUploadManager[] => {
-    return this.chatAdapter.registerActiveFileUploads(files);
+    return this.chatAdapter?.registerActiveFileUploads(files) ?? [];
   };
   /* @conditional-compile-remove(attachment-download) @conditional-compile-remove(attachment-upload) */
   public registerCompletedFileUploads = (metadata: AttachmentMetadata[]): FileUploadManager[] => {
-    return this.chatAdapter.registerCompletedFileUploads(metadata);
+    return this.chatAdapter?.registerCompletedFileUploads(metadata) ?? [];
   };
   /* @conditional-compile-remove(attachment-download) @conditional-compile-remove(attachment-upload) */
   public clearFileUploads = (): void => {
-    this.chatAdapter.clearFileUploads();
+    this.chatAdapterPromise.then((adapter) => {
```

Comment:
void functions can't have their internals wrapped in an async block. If I'm a developer calling a synchronous method, it can't suddenly become asynchronous across releases. The developer could be reliant on the subsequent code being assertive that the previous statement has finished execution.
We need to throw an error here if the chatadapter hasn't been created already

Reply:
Alternatively we only use the promise logic _iff_ we haven't resolved chatAdapter previously, meaning we don't break any customers today.
i.e.
```ts
public clearFileUploads = (): void => {
    if (this.chatAdapter) {
       this.chatAdapter.clearFileUploads();
    } else {
      this.chatAdapterPromise.then((adapter) => {
         adapter.clearFileUploads();
      }
    }
}
```

Which would get wrapped in a function to reuse across all these functions, something like:
```ts
public clearFileUploads(): void {
  return this.executeWithResolvedChatAdapter(clearFileUploads);
}

private executeWithResolvedChatAdapter(callback: (adapter: ChatAdapter) => void): void {
  if (this.chatAdapter) {
    callback(this.chatAdapter);
  } else {
    this.chatAdapterPromise.then((adapter) => {
      return callback(adapter);
    });
  }
}
```